sqlmesh/utils/metaprogramming.py

```diff
@@ -440,26 +439,34 @@ def print_exception(
     Args:
         exception: The exception to print the stack trace for.
         python_env: The environment containing stringified python code.
-        out: The output stream to write to.
     """
     tb: t.List[str] = []
+    indent = ""
 
     for error_line in format_exception(exception):
-        match = re.search('File "<string>", line (.*), in (.*)', error_line)
+        traceback_match = error_line.startswith("Traceback (most recent call last):")
```

Comment:
[Nit] We don't need to match ll of these at once since we'd only use a handful. A better approach would be:
```
traceback_match = error_line.startswith("Traceback (most recent call last):")
model_def_match = re.search('File ".*?core/model/definition.py', error_line)
if traceback_match or model_def_match:
    continue
    
error_match = re.search("^.*?Error: ", error_line)
if error_match:
    tb.append(f"{indent*2}  {error_line}")
    continue
...
```